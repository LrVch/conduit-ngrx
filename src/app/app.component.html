<div [class]="'theme-wrapper ' + (effectiveTheme$ | async)"
     [ngClass]="{'sticky-header': (stickyHeader$ | async)}">
  <mat-sidenav-container>
    <mat-sidenav [opened]="sideNavOpen$ | async"
                 (closed)="onCloseSideNave()"
                 [mode]="asideOpenMode$ | async">
      <app-aside *ngIf="(settings$ | async)"
                 [theme]="theme$ | async"
                 [themes]="themes$ | async"
                 [stickyHeader]="stickyHeader$ | async"
                 [routeAnimationsEnabled]="routeAnimationsEnabled$ | async"
                 [asideOpenMode]="asideOpenMode$ | async"
                 [asideOpenModes]="asideOpenModes$ | async"
                 [autoNightMode]="autoNightMode$ | async"
                 [autoNightModeFrom]="autoNightModeFrom$ | async"
                 [autoNightModeTo]="autoNightModeTo$ | async"
                 [routeAnimateonsChangeType]="routeAnimateonsChangeType$ | async"
                 [routeAnimateonsChangeTypes]="routeAnimateonsChangeTypes$ | async"
                 (changeRouteAnimationsType)="onChangeRouteAnimationsType($event)"
                 (changeStickyHeader)="onChangeStickyHeader($event)"
                 (changeAsideOpenMode)="onChangeAsideOpenMode($event)"
                 (changeAutoNightMode)="onChangeAutoNightMode($event)"
                 (changeAutoNightModeFrom)="onChangeAutoNightModeFrom($event)"
                 (changeAutoNightModeTo)="onCangeAutoNightModeTo($event)"
                 (changeRouteAnimationsEnable)="onChangeRouteAnimationsEnable($event)"
                 (changeTheme)="onChangeTheme($event)"></app-aside>
    </mat-sidenav>
    <mat-sidenav-content>

      <div class="layout">
        <div class="main">
          <div class="main-content">
            <app-header [user]="user$ | async"
                        [loggedIn]="loggedIn$ | async"
                        [showLoader]="showMailLoader$ | async"
                        [languages]="languages$ | async"
                        [language]="language$ | async"
                        (changeLanguage)="onChangeLanguage($event)"
                        (toggleSideNave)="onToggleSideNave()"></app-header>

            <div [@routeAnimation]="prepRouteState(routerOutlet)">
              <router-outlet #routerOutlet="outlet"></router-outlet>
            </div>
          </div>

          <app-footer></app-footer>
        </div>
      </div>
    </mat-sidenav-content>

  </mat-sidenav-container>

</div>