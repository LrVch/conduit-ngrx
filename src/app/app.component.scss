// @import '../styles-variables';

.theme-wrapper {
  width: 100%;
  height: 100%;
}

mat-sidenav-container {
  height: 100%;
  width: 100%;
}

mat-sidenav {
  width: 250px;

  // .mat-nav-list {
  //   padding-top: 0;
  // }
}