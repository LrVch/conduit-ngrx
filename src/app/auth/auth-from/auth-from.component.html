<form [formGroup]="form"
      [appAccentOnInvalidFromField]="form"
      (ngSubmit)="submit()">

  <mat-form-field *ngIf="authType === 'register'"
                  class="input-full-width">
    <input matInput
           formControlName="username"
           placeholder="Username"
           autocomplete="on"
           type="text">
    <mat-error *ngIf="requiredUsername">
      Username is <strong>required</strong>
    </mat-error>
  </mat-form-field>

  <mat-form-field class="input-full-width">
    <input matInput
           formControlName="email"
           placeholder="Email"
           type="text">
    <mat-error *ngIf="invalidEmail">
      Please enter a valid email address
    </mat-error>
    <mat-error *ngIf="requiredEmail">
      Email is <strong>required</strong>
    </mat-error>
  </mat-form-field>

  <mat-form-field class="input-full-width">
    <input matInput
           formControlName="password"
           placeholder="Password"
           [attr.autocomplete]="authType === 'register' ? 'new-password': 'on'"
           type="password">
    <mat-error *ngIf="requiredPassword">
      Password is <strong>required</strong>
    </mat-error>
    <mat-error *ngIf="minLengthPassword">
      Password cannot be less than <strong>{{ minLengthPasswordCharsLength }}</strong> characters
    </mat-error>
    <app-password-strength *ngIf="authType === 'register'"
                           [password]="$passValue | async"></app-password-strength>
  </mat-form-field>
  
  <br>
  <br>
  <button type="submit"
          [disabled]="submitting"
          mat-flat-button
          color="primary">Login</button>
  <br>
  <br>

</form>