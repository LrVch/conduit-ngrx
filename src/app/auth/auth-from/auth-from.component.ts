import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { FormGroup, FormBuilder, Validators, FormControl } from '@angular/forms';
import { Credentials } from 'src/app/core/models/credentials.model';

export interface AuthPayload {
  credentials: Credentials;
  authType: string;
}

@Component({
  selector: 'app-auth-from',
  templateUrl: './auth-from.component.html'
})

export class AuthFromComponent implements OnInit {
  @Input('authType') authType: string;
  @Output() submitForm = new EventEmitter<AuthPayload>();
  @Input() set isSubmitting(isSubmitting: boolean) {
    this._isSubmitting = isSubmitting;
    this.toggleDisableFields(this.form, isSubmitting);
  }

  _isSubmitting: boolean;

  form: FormGroup;

  constructor(
    private fb: FormBuilder,
  ) { }

  ngOnInit() {
    this.form = this.fb.group({
      'email': ['', Validators.required],
      'password': ['', Validators.required]
    });

    if (this.authType === 'register') {
      this.form.addControl('username', new FormControl('', Validators.required));
    }
  }

  submit() {
    if (this.form.valid) {
      const authPayload: AuthPayload = {
        credentials: this.form.value,
        authType: this.authType
      };

      this.submitForm.emit(authPayload);
    } else {
      this.validateAllFormFields(this.form);
    }
  }

  validateAllFormFields(formGroup: FormGroup): void {
    Object.keys(formGroup.controls).forEach(field => {
      const control = formGroup.get(field);
      if (control instanceof FormControl) {
        control.markAsTouched({ onlySelf: true });
      } else if (control instanceof FormGroup) {
        this.validateAllFormFields(control);
      }
    });
  }

  toggleDisableFields(formGroup: FormGroup, disable: boolean): void {
    if (!formGroup) {
      return;
    }
    Object.keys(formGroup.controls).forEach(field => {
      const control = formGroup.get(field);
      if (control instanceof FormControl) {
        if (disable) {
          control.disable();
        } else {
          control.enable();
        }
      } else if (control instanceof FormGroup) {
        this.validateAllFormFields(control);
      }
    });
  }
}
