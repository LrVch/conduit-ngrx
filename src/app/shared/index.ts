export * from './article/article.component';
export * from './articles-list/articles-list.component';
export * from './content-tabs/content-tabs.component';
export * from './tags/tags.component';
export * from './favorite/favorite.component';
export * from './layout';
export * from './shared.module';
export * from './utils';
export * from './list-error/list-error.component';
export * from './show-authed.directive';
export * from './user-info/user-info.component';
export * from './follow-button/follow-button.component';
export * from './tag-list/tag-list.component';
export * from './comments/comments.component';
export * from './base-from/base-from.component';
