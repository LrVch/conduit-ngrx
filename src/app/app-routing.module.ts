import { NgModule } from '@angular/core';
import { Routes, RouterModule, PreloadAllModules } from '@angular/router';
import { PageNotFoundComponent } from './shared';
import { AuthGuard } from './core';
const routes: Routes = [
  {
    path: 'settings',
    loadChildren: './settings/settings.module#SettingsModule',
    canLoad: [AuthGuard],
  },
  {
    path: 'profile',
    loadChildren: './profile/profile.module#ProfileModule',
  },
  {
    path: 'editor',
    loadChildren: './editor/editor.module#EditorModule',
    canLoad: [AuthGuard],
  },
  {
    path: 'article',
    loadChildren: './article/article.module#ArticleModule',
  },
  {
    path: '**',
    component: PageNotFoundComponent,
    data: { title: 'conduit.menu.notfound' }
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes, {
    scrollPositionRestoration: 'enabled'
    // preload all modules; optionally we could
    // implement a custom preloading strategy for just some
    // of the modules (PRs welcome ðŸ˜‰)
    // preloadingStrategy: PreloadAllModules
  })],
  exports: [RouterModule]
})
export class AppRoutingModule { }
