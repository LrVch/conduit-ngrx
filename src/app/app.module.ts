import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import 'hammerjs';


/* Common */
import { AppComponent } from './app.component';
import { LayoutModule } from './layout/layout.module';
import { CoreModule, } from './core/core.module';
import {
  SharedModule,
  HeaderComponent,
  FooterComponent
} from './shared';
import { HomeModule } from './home/home.module';
import { ArticlesModule } from './articles/articles.module';
import { AppRoutingModule } from './app-routing.module';
import { AuthModule } from './auth/auth.module';


@NgModule({
  declarations: [
    AppComponent,
    HeaderComponent,
    FooterComponent,
  ],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,

    CoreModule,
    SharedModule,
    LayoutModule,
    AuthModule,
    HomeModule,
    ArticlesModule,

    AppRoutingModule,
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }
