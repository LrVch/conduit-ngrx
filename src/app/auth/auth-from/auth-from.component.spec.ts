import { async, ComponentFixture } from '@angular/core/testing';

import { ConfigureFn, configureTests, getUser } from '../../lib/testing';
import { AuthFromComponent } from './auth-from.component';
import { DebugElement } from '@angular/core';
import { ReactiveFormsModule, FormsModule } from '@angular/forms';
import { ScrollService, DomUtilService } from 'src/app/core';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { By } from '@angular/platform-browser';
import { PasswordStrengthComponent, MaterialModule, AccentOnInvalidFromFieldDirective } from '../../shared';

describe('AuthFromComponent', () => {
    let component: AuthFromComponent;
    let fixture: ComponentFixture<AuthFromComponent>;
    let de: DebugElement;
    let user;

    beforeEach(
        async(() => {
            user = getUser();
            const configure: ConfigureFn = testBed => {
                testBed.configureTestingModule({
                    declarations: [AuthFromComponent, AccentOnInvalidFromFieldDirective, PasswordStrengthComponent],
                    imports: [FormsModule, ReactiveFormsModule, MaterialModule, NoopAnimationsModule],
                    providers: [ScrollService, DomUtilService]
                });
            };

            configureTests(configure).then(testBed => {
                fixture = testBed.createComponent(AuthFromComponent);
                component = fixture.componentInstance;
                de = fixture.debugElement;
                fixture.detectChanges();
            });
        })
    );

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should has "submitting" props', () => {
        expect(component.submitting).toBe(false);
    });

    describe('login form', () => {
        it('should create from with 2 controls', () => {
            const form = component.form;

            expect(form.contains('email')).toBeTruthy();
            expect(form.contains('password')).toBeTruthy();
        });

        it('email control should be required', () => {
            const form = component.form;
            const email = form.get('email');

            expect(email.valid).toBeFalsy();
        });

        it('email control should be has email validator', () => {
            const form = component.form;
            const email = form.get('email');

            email.setValue('not real email');

            expect(email.valid).toBeFalsy();

            email.setValue('reale@email.ai');

            expect(email.valid).toBeTruthy();
        });

        it('password control should be required', () => {
            const form = component.form;
            const password = form.get('password');

            expect(password.valid).toBeFalsy();
        });
    });

    describe('register form', () => {
        it('should create from with 3 controls', () => {
            component.authType = 'register';
            component.ngOnInit();

            const form = component.form;

            expect(form.contains('email')).toBeTruthy();
            expect(form.contains('password')).toBeTruthy();
            expect(form.contains('username')).toBeTruthy();
        });

        it('email control should be required', () => {
            component.authType = 'register';
            component.ngOnInit();

            const form = component.form;

            const email = form.get('email');

            expect(email.valid).toBeFalsy();
        });

        it('email control should be has email validator', () => {
            component.authType = 'register';
            component.ngOnInit();

            const form = component.form;

            const email = form.get('email');

            email.setValue('not real email');

            expect(email.valid).toBeFalsy();

            email.setValue('reale@email.ai');

            expect(email.valid).toBeTruthy();
        });

        it('password control should be required', () => {
            component.authType = 'register';
            component.ngOnInit();

            const form = component.form;

            const password = form.get('password');

            expect(password.valid).toBeFalsy();
        });

        it('password control should be at least 8 characters required', () => {
            component.authType = 'register';
            component.ngOnInit();

            const form = component.form;

            const password = form.get('password');

            password.setValue('1234567');

            expect(password.valid).toBeFalsy();

            password.setValue('12345678');

            expect(password.valid).toBeTruthy();
        });

        it('username control should be required', () => {
            component.authType = 'register';
            component.ngOnInit();

            const form = component.form;

            const username = form.get('username');

            expect(username.valid).toBeFalsy();
        });
    });

    it('should show error message', () => {
        component.form.get('password').setValue('');
        component.form.get('email').setValue('');

        component.submit();

        fixture.detectChanges();

        (<any>expect(fixture)).toMatchSnapshot();
    });

    it('should disable button', () => {
        component.isSubmitting = true;
        fixture.detectChanges();
        (<any>expect(fixture)).toMatchSnapshot();
    });

    it('should show password strength compenent if form type is register', () => {
        component.authType = 'register';
        component.ngOnInit();
        fixture.detectChanges();
        (<any>expect(fixture)).toMatchSnapshot();
    });

    it('should disable form', () => {
        component.isSubmitting = true;

        expect(component.form.status).toBe('DISABLED');
    });

    it('should raise "submitForm" when form valid on submit event and pass user data', () => {
        let email = '';
        let password = '';
        const expectedEmail = 'real@email.ai';
        const expectedPassword = '12345678';

        component.form.get('email').setValue(expectedEmail);
        component.form.get('password').setValue(expectedPassword);

        component.submitForm.subscribe(res => {
            email = res.credentials.email;
            password = res.credentials.password;
        });

        fixture.detectChanges();
        const formDe = de.query(By.css('form'));

        formDe.triggerEventHandler('ngSubmit', null);

        expect(email).toBe(expectedEmail);
        expect(password).toBe(expectedPassword);
    });

    it('shouldn\'t raise "submitComment" when form invalid', () => {
        let wasEmited = false;

        component.submitForm.subscribe(c => wasEmited = true);

        fixture.detectChanges();
        const formDe = de.query(By.css('form'));

        formDe.triggerEventHandler('ngSubmit', null);

        expect(wasEmited).toBe(false);
    });

    it('should invoke validateAllFormFields method when form invalid', () => {
        const spy = jest.spyOn(component, 'validateAllFormFields');

        fixture.detectChanges();
        const formDe = de.query(By.css('form'));
        formDe.triggerEventHandler('ngSubmit', null);

        expect(spy).toHaveBeenCalled();
    });
});
