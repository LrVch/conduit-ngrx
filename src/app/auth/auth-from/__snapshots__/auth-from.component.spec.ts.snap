// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthFromComponent should disable button 1`] = `
<app-auth-from
  $passValue={[Function EventEmitter]}
  fb={[Function FormBuilder]}
  form={[Function FormGroup]}
  submitForm={[Function EventEmitter]}
  submitting={[Function Boolean]}
>
  <form
    class="ng-untouched ng-pristine"
    ng-reflect-app-accent-on-invalid-from-field="[object Object]"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    
    <mat-form-field
      class="input-full-width mat-form-field ng-tns-c26-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine _mat-animation-noopable mat-form-field-disabled"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-infix"
          >
            <input
              aria-invalid="false"
              aria-required="false"
              class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine"
              disabled=""
              formcontrolname="email"
              id="mat-input-26"
              matinput=""
              ng-reflect-name="email"
              ng-reflect-placeholder="Email"
              ng-reflect-type="text"
              placeholder="Email"
              type="text"
            />
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                aria-owns="mat-input-26"
                class="mat-form-field-label ng-tns-c26-0 mat-empty mat-form-field-empty ng-star-inserted"
                for="mat-input-26"
                id="mat-form-field-label-53"
                ng-reflect-disabled="true"
                ng-reflect-ng-switch="false"
              >
                
                
                Email
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-underline ng-tns-c26-0 ng-star-inserted"
        >
          <span
            class="mat-form-field-ripple"
          />
        </div>
        <div
          class="mat-form-field-subscript-wrapper"
          ng-reflect-ng-switch="hint"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-tns-c26-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
    <mat-form-field
      class="input-full-width mat-form-field ng-tns-c26-1 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine _mat-animation-noopable mat-form-field-disabled"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-infix"
          >
            <input
              aria-invalid="false"
              aria-required="false"
              autocomplete="on"
              class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine"
              disabled=""
              formcontrolname="password"
              id="mat-input-27"
              matinput=""
              ng-reflect-name="password"
              ng-reflect-placeholder="Password"
              ng-reflect-type="password"
              placeholder="Password"
              type="password"
            />
            
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                aria-owns="mat-input-27"
                class="mat-form-field-label ng-tns-c26-1 mat-empty mat-form-field-empty ng-star-inserted"
                for="mat-input-27"
                id="mat-form-field-label-55"
                ng-reflect-disabled="true"
                ng-reflect-ng-switch="false"
              >
                
                
                Password
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-underline ng-tns-c26-1 ng-star-inserted"
        >
          <span
            class="mat-form-field-ripple"
          />
        </div>
        <div
          class="mat-form-field-subscript-wrapper"
          ng-reflect-ng-switch="hint"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-tns-c26-1 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
    <br />
    <br />
    <button
      class="mat-flat-button mat-primary _mat-animation-noopable"
      color="primary"
      disabled=""
      mat-flat-button=""
      ng-reflect-color="primary"
      ng-reflect-disabled="true"
      type="submit"
    >
      <span
        class="mat-button-wrapper"
      >
        Login
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="true"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    <br />
    <br />
  </form>
</app-auth-from>
`;

exports[`AuthFromComponent should show error message 1`] = `
<app-auth-from
  $passValue={[Function EventEmitter]}
  fb={[Function FormBuilder]}
  form={[Function FormGroup]}
  submitForm={[Function EventEmitter]}
  submitting="false"
>
  <form
    class="ng-untouched ng-pristine ng-invalid"
    ng-reflect-app-accent-on-invalid-from-field="[object Object]"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    
    <mat-form-field
      class="input-full-width mat-form-field ng-tns-c24-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-pristine ng-invalid _mat-animation-noopable mat-form-field-invalid ng-touched"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-infix"
          >
            <input
              aria-describedby="mat-error-24"
              aria-invalid="true"
              aria-required="false"
              class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-pristine ng-invalid ng-touched"
              formcontrolname="email"
              id="mat-input-24"
              matinput=""
              ng-reflect-name="email"
              ng-reflect-placeholder="Email"
              ng-reflect-type="text"
              placeholder="Email"
              type="text"
            />
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                aria-owns="mat-input-24"
                class="mat-form-field-label ng-tns-c24-0 mat-empty mat-form-field-empty ng-star-inserted"
                for="mat-input-24"
                id="mat-form-field-label-49"
                ng-reflect-disabled="true"
                ng-reflect-ng-switch="false"
              >
                
                
                Email
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-underline ng-tns-c24-0 ng-star-inserted"
        >
          <span
            class="mat-form-field-ripple"
          />
        </div>
        <div
          class="mat-form-field-subscript-wrapper"
          ng-reflect-ng-switch="error"
        >
          
          <div
            class="ng-tns-c24-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted ng-animating"
            style=""
          >
            
            
            <mat-error
              class="mat-error ng-star-inserted"
              id="mat-error-24"
              role="alert"
            >
               Email is 
              <strong>
                required
              </strong>
            </mat-error>
          </div>
          
        </div>
      </div>
    </mat-form-field>
    <mat-form-field
      class="input-full-width mat-form-field ng-tns-c24-1 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-pristine ng-invalid _mat-animation-noopable mat-form-field-invalid ng-touched"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-infix"
          >
            <input
              aria-describedby="mat-error-25"
              aria-invalid="true"
              aria-required="false"
              autocomplete="on"
              class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-pristine ng-invalid ng-touched"
              formcontrolname="password"
              id="mat-input-25"
              matinput=""
              ng-reflect-name="password"
              ng-reflect-placeholder="Password"
              ng-reflect-type="password"
              placeholder="Password"
              type="password"
            />
            
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                aria-owns="mat-input-25"
                class="mat-form-field-label ng-tns-c24-1 mat-empty mat-form-field-empty ng-star-inserted"
                for="mat-input-25"
                id="mat-form-field-label-51"
                ng-reflect-disabled="true"
                ng-reflect-ng-switch="false"
              >
                
                
                Password
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-underline ng-tns-c24-1 ng-star-inserted"
        >
          <span
            class="mat-form-field-ripple"
          />
        </div>
        <div
          class="mat-form-field-subscript-wrapper"
          ng-reflect-ng-switch="error"
        >
          
          <div
            class="ng-tns-c24-1 ng-trigger ng-trigger-transitionMessages ng-star-inserted ng-animating"
            style=""
          >
            
            <mat-error
              class="mat-error ng-star-inserted"
              id="mat-error-25"
              role="alert"
            >
               Password is 
              <strong>
                required
              </strong>
            </mat-error>
            
          </div>
          
        </div>
      </div>
    </mat-form-field>
    <br />
    <br />
    <button
      class="mat-flat-button mat-primary _mat-animation-noopable"
      color="primary"
      mat-flat-button=""
      ng-reflect-color="primary"
      ng-reflect-disabled="false"
      type="submit"
    >
      <span
        class="mat-button-wrapper"
      >
        Login
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    <br />
    <br />
  </form>
</app-auth-from>
`;

exports[`AuthFromComponent should show password strength compenent if form type is register 1`] = `
<app-auth-from
  $passValue={[Function EventEmitter]}
  authType={[Function String]}
  fb={[Function FormBuilder]}
  form={[Function FormGroup]}
  submitForm={[Function EventEmitter]}
  submitting="false"
>
  <form
    class="ng-untouched ng-pristine ng-invalid"
    ng-reflect-app-accent-on-invalid-from-field="[object Object]"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    
    <mat-form-field
      class="input-full-width mat-form-field ng-tns-c28-2 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid _mat-animation-noopable ng-star-inserted"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-infix"
          >
            <input
              aria-invalid="false"
              aria-required="false"
              autocomplete="on"
              class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-invalid"
              formcontrolname="username"
              id="mat-input-30"
              matinput=""
              ng-reflect-name="username"
              ng-reflect-placeholder="Username"
              ng-reflect-type="text"
              placeholder="Username"
              type="text"
            />
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                aria-owns="mat-input-30"
                class="mat-form-field-label ng-tns-c28-2 mat-empty mat-form-field-empty ng-star-inserted"
                for="mat-input-30"
                id="mat-form-field-label-61"
                ng-reflect-disabled="true"
                ng-reflect-ng-switch="false"
              >
                
                
                Username
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-underline ng-tns-c28-2 ng-star-inserted"
        >
          <span
            class="mat-form-field-ripple"
          />
        </div>
        <div
          class="mat-form-field-subscript-wrapper"
          ng-reflect-ng-switch="hint"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-tns-c28-2 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
    <mat-form-field
      class="input-full-width mat-form-field ng-tns-c28-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid _mat-animation-noopable"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-infix"
          >
            <input
              aria-invalid="false"
              aria-required="false"
              class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-invalid"
              formcontrolname="email"
              id="mat-input-28"
              matinput=""
              ng-reflect-name="email"
              ng-reflect-placeholder="Email"
              ng-reflect-type="text"
              placeholder="Email"
              type="text"
            />
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                aria-owns="mat-input-28"
                class="mat-form-field-label ng-tns-c28-0 mat-empty mat-form-field-empty ng-star-inserted"
                for="mat-input-28"
                id="mat-form-field-label-57"
                ng-reflect-disabled="true"
                ng-reflect-ng-switch="false"
              >
                
                
                Email
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-underline ng-tns-c28-0 ng-star-inserted"
        >
          <span
            class="mat-form-field-ripple"
          />
        </div>
        <div
          class="mat-form-field-subscript-wrapper"
          ng-reflect-ng-switch="hint"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-tns-c28-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
    <mat-form-field
      class="input-full-width mat-form-field ng-tns-c28-1 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid _mat-animation-noopable"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-infix"
          >
            <input
              aria-invalid="false"
              aria-required="false"
              autocomplete="new-password"
              class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-invalid"
              formcontrolname="password"
              id="mat-input-29"
              matinput=""
              ng-reflect-name="password"
              ng-reflect-placeholder="Password"
              ng-reflect-type="password"
              placeholder="Password"
              type="password"
            />
            
            <app-password-strength
              class="ng-star-inserted"
            >
              <div
                class="password-strength"
                ng-reflect-klass="password-strength"
                ng-reflect-ng-class="[object Object]"
              >
                
              </div>
            </app-password-strength>
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                aria-owns="mat-input-29"
                class="mat-form-field-label ng-tns-c28-1 mat-empty mat-form-field-empty ng-star-inserted"
                for="mat-input-29"
                id="mat-form-field-label-59"
                ng-reflect-disabled="true"
                ng-reflect-ng-switch="false"
              >
                
                
                Password
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-underline ng-tns-c28-1 ng-star-inserted"
        >
          <span
            class="mat-form-field-ripple"
          />
        </div>
        <div
          class="mat-form-field-subscript-wrapper"
          ng-reflect-ng-switch="hint"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-tns-c28-1 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
    <br />
    <br />
    <button
      class="mat-flat-button mat-primary _mat-animation-noopable"
      color="primary"
      mat-flat-button=""
      ng-reflect-color="primary"
      ng-reflect-disabled="false"
      type="submit"
    >
      <span
        class="mat-button-wrapper"
      >
        Login
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    <br />
    <br />
  </form>
</app-auth-from>
`;
